File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/Specifications/Commands/SelectSpecificationCommand.js
SIE.defineCommand('SIE.Web.MES.TaskManagement.Specifications.Commands.SelectSpecificationCommand', {
    extend: 'SIE.cmd.LookupCommandBase',
    userConfig: {
        dataParams: { specKeyPrototyName: 'SpecificationId', targetClassName: 'SIE.MES.TaskManagement.Specifications.Specification' },
    },
    meta: { text: "选择", group: "edit", iconCls: "icon-PlaylistCheck icon-blue" },
    save: function (win) {
        /// <summary>
        /// 保存选择的操作列表。
        /// </summary>
        var me = this;
        /* post数据结构*/
        var indata = {};
        /* post数据结构*/
        var selections = this._targetView.getSelection();
        if (selections && selections.length > 0) {
            var operationDatas = [];
            SIE.each(selections, function (item) {
                var specificationId = item.getId();
                if (me._sourceViewSelectItems.indexOf(specificationId) === -1) {
                    var productSpecificationDetail = { ProductSpecificationId: me._sourceId, SpecificationId: specificationId };
                    operationDatas.push(productSpecificationDetail);
                }
            });
            indata = operationDatas;
            me._targetView.execute({
                data: indata,
                success: function (res) {
                    win.close();  //关闭模态窗口
                    me._ownerView.loadChildData(true); //重载视图数据
                }
            }, me._ownerView);
        }
        else {
            SIE.Msg.showWarning('没有可提交的数据'.t());
        }
    }
    // end 
});

File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/Specifications/Commands/SelectSpecificationCommand.cs
using SIE.Domain;
using SIE.MES.TaskManagement.Specifications;
using SIE.Web.Command;
using System;
using System.Collections.Generic;

namespace SIE.Web.MES.TaskManagement.Specifications.Commands
{
    /// <summary>
    /// 规格件选择
    /// </summary>
    public class SelectSpecificationCommand : ViewCommand
    {
        /// <summary>
        /// 执行选择
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var meta = ClientEntities.Find("SIE.MES.TaskManagement.Specifications.ProductSpecificationDetail");
            var savedData = RF.Find(meta.EntityType).NewList();
            var prodSpecificationDetails = args.Data.ToJsonObject<List<ProductSpecificationDetail>>();
            Check.NotNullOrEmpty(prodSpecificationDetails, nameof(prodSpecificationDetails));
            if (null == prodSpecificationDetails || prodSpecificationDetails.Count == 0)
            {
                throw new ArgumentNullException("{0}数据参数不能为空".L10nFormat(nameof(prodSpecificationDetails)));
            }
            foreach (var item in prodSpecificationDetails)
            {
                var prodSpecificationDetail = new ProductSpecificationDetail();
                prodSpecificationDetail.SpecificationId = item.SpecificationId;
                prodSpecificationDetail.ProductSpecificationId = item.ProductSpecificationId;
                prodSpecificationDetail.Qty = 1;
                savedData.Add(prodSpecificationDetail);
            }
            RF.Save(savedData);

            return true;
        }
    }
}
